### PR TITLE
refactor: use optional chaining for improved readability and maintained functionality

### PR DESCRIPTION
**Pull request by thebedigupta**
**Description**
In this PR I worked on this issue [link](https://sonarcloud.io/project/issues?open=AZB4sejpgviDxNeZYbKV&id=asyncapi_generator)

I already added a test file where I captured all the tests that I performed [test-results.txt](https://github.com/user-attachments/files/17248490/test-results.txt)

- Simplified Null Check
- Maintained functionality

**This change addresses suggestions from SonarCloud to adopt modern JavaScript features.**

**Related issue(s)**

### CODE REVIEW

File: apps/generator/cli.js
Comment (DhairyaMajmudar) [inline apps/generator/cli.js:1]: Get back this file pls.

File: apps/generator/lib/generator.js
Comment (DhairyaMajmudar) [inline apps/generator/lib/generator.js:1]: Remove unwanted comments from this file

File: apps/generator/lib/generator.js [hunk 1]
```diff
@@ -130,17 +130,32 @@ class Generator {
...
-      const self = this;
       Object.defineProperty(this.templateParams, key, {
         enumerable: true,
-        get() {
-          if (!self.templateConfig.parameters || !self.templateConfig.parameters[key]) {
-            throw new Error(`Template parameter "${key}" has not been defined in the package.json file under generator property. Please make sure it's listed there before you use it in your template.`);
+        get: () => {
+          // Using ?. to check if parameters and the key exist
+          // This is shorter and does the same thing as the old if statements
+          if (this.templateConfig.parameters?.[key] == null) {
```
Comment (DhairyaMajmudar) [inline apps/generator/lib/generator.js:138]: Here there's one condition also exists `!self.templateConfig.parameters` pls. add that also in validation.

Comment (thebedigupta) [reply apps/generator/lib/generator.js]: Please check PR description where I mention the issue that I solved in this PR sonarcloud highlight the issue that Prefer using an optional chain expression instead, as it's more concise and easier to read. that's what I did if i implement one more codition `!self.templateConfig.parameters` then this PR makes no sense here I hope you understand 

This change makes the code more concise and easier to read while maintaining the same functionality. It checks if either the parameters object doesn't exist or if the specific key is not defined within it.

**Review by derberg — COMMENTED** [review]
please remove any changes to `cli.js` and any formatting changes you added to your PR

**Comment by thebedigupta** [comment]
> please remove any changes to `cli.js` and any formatting changes you added to your PR

I  removed all kind of changes in the `cli.js` file

**Review by derberg — COMMENTED** [review]
`cli.js` is still showing up in the list of modified files

**Comment by DhairyaMajmudar** [comment]
Pls. have a look on the comment https://github.com/asyncapi/generator/pull/1283#issuecomment-2404066707 to fix this

**Pull request closed without merging**